radix-engine-stores/src/rocks_db.rs

```diff
@@ -62,6 +62,22 @@ impl CommittableSubstateDatabase for RocksdbSubstateStore {
     }
 }
 
+impl ListableSubstateDatabase for RocksdbSubstateStore {
+    fn list_partition_keys(&self) -> Box<dyn Iterator<Item = DbPartitionKey> + '_> {
+        let partition_keys: Vec<DbPartitionKey> = self
+            .db
+            .iterator(IteratorMode::Start)
+            .map(|kv| {
+                let (iter_key_bytes, _) = kv.as_ref().unwrap();
+                let (iter_key, _) = decode_from_rocksdb_bytes(iter_key_bytes);
```

Comment:
hah, funny that `ListableSubstateDatabase` existed but had no impl :)

Anyway, its contract is not clear, but I assume it wants to list every partition key once. AFAIU, `InMemorySubstateDatabase` does it this way, but the rocksdb here lists duplicates?

Reply:
Indeed keys might be duplicated. Fixed.
Decided not to use `dedup()` to not change the order of the keys.